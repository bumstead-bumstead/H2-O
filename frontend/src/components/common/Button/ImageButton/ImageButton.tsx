import type { ButtonHTMLAttributes } from 'react';
import styled from '@emotion/styled';

type Props = ButtonHTMLAttributes<HTMLButtonElement>;

function ImageButton({ children, ...restProps }: Props) {
  return <StyledButton {...restProps}>{children}</StyledButton>;
}

export default ImageButton;

const StyledButton = styled.button<Props>`
  ${({ theme }) => theme.typography.TextKRMedium12}
  color: white;
  background-color: #757575;
  border-radius: 20px;
  opacity: 0.5;
  width: 76px;
  height: 28px;
`;
