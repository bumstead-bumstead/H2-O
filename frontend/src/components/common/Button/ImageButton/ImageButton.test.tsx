import ImageButton from './ImageButton';
import { render, screen, userEvent, waitFor } from '@/tests/test-util';

describe('CTAButton 컴포넌트 테스트', () => {
  it('버튼이 정상적으로 렌더링된다.', () => {
    render(<ImageButton>버튼</ImageButton>);

    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('버튼의 onClick 핸들러가 클릭시 정상적으로 호출된다.', async () => {
    const mockOnClick = vi.fn();
    render(<ImageButton onClick={mockOnClick}>버튼</ImageButton>);

    userEvent.click(screen.getByRole('button'));
    await waitFor(() => expect(mockOnClick).toBeCalledTimes(1));
  });

  it('버튼에 disabled 속성을 전달하면 disabled 된다.', async () => {
    const mockOnClick = vi.fn();
    render(
      <ImageButton disabled onClick={mockOnClick}>
        버튼
      </ImageButton>,
    );

    const button = screen.getByRole('button');
    expect(button).toBeDisabled();

    userEvent.click(button);
    await waitFor(() => expect(mockOnClick).not.toBeCalled());
  });

  describe('ImageButton 스냅샷 테스트', () => {
    it('버튼의 스냅샷이 일치한다.', () => {
      render(<ImageButton>버튼</ImageButton>);

      expect(screen.getByRole('button')).toMatchSnapshot();
    });
  });
});
