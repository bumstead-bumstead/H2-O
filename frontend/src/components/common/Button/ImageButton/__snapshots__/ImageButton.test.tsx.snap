// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CTAButton 컴포넌트 테스트 > ImageButton 스냅샷 테스트 > 버튼의 스냅샷이 일치한다. 1`] = `
.emotion-0 {
  font-family: Hyundai Sans Text KR,sans-serif;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  letter-spacing: -0.03em;
  color: white;
  background-color: #757575;
  border-radius: 20px;
  opacity: 0.5;
  width: 76px;
  height: 28px;
}

<button
  class="emotion-0"
>
  버튼
</button>
`;
